src/core/ddsi/src/q_init.c

```diff
@@ -1164,7 +1163,7 @@ static int iceoryx_init (struct ddsi_domaingv *gv)
     gv->loc_iceoryx_addr.kind = NN_LOCATOR_KIND_SHEM;
     gv->loc_iceoryx_addr.port = 0;
     memset ((char *) gv->loc_iceoryx_addr.address, 0, sizeof (gv->loc_iceoryx_addr.address));
-    ddsrt_strlcpy ((char *) gv->loc_iceoryx_addr.address, sptr, strlen (sptr));
+    ddsrt_strlcpy ((char *) gv->loc_iceoryx_addr.address, sptr, strlen (sptr) + 1);
```

Comment:
Interesting find! The original code would've replaced the final 4 bits of the MAC address by 0.

That said, I have two issues with the code here:
* the original as well as the new code abuses `ddsrt_strlcpy`, because the third parameter really should be the size of the destination;
* locators are not strings anyway, so I think the conversion to string is quite useless.

I think it would make more sense altogether to do:
```C
    memset (gv->loc_iceoryx_addr.address, 0, sizeof (gv->loc_iceoryx_addr.address));
    if (ddsrt_eth_get_mac_addr (gv->interfaces[if_index].name, gv->loc_iceoryx_addr.address))
    {
      GVERROR ("Unable to get MAC address for iceoryx\n");
      return -1;
    }
    gv->loc_iceoryx_addr.kind = NN_LOCATOR_KIND_SHEM;
    gv->loc_iceoryx_addr.port = 0;
```